### PR TITLE
Replace usage of transform with updateTransforms

### PR DESCRIPTION
**Pull request by thebrianchen**
Porting from [web](https://github.com/firebase/firebase-js-sdk/pull/4153).

### CODE REVIEW

File: firebase-firestore/src/test/java/com/google/firebase/firestore/remote/RemoteSerializerTest.java [hunk 1]
```diff
@@ -378,64 +378,98 @@ public void testEncodesPatchMutationWithFieldMask() {
...
+                    .setSetToServerValue(ServerValue.REQUEST_TIME))
+            .addUpdateTransforms(
+                FieldTransform.newBuilder()
+                    .setFieldPath("bar.baz")
+                    .setSetToServerValue(ServerValue.REQUEST_TIME))
             .setCurrentDocument(Precondition.newBuilder().setExists(true))
             .build();
+    assertRoundTripForMutation(mutation, expected);
 
+    mutation =
```
Comment (schmidt-sebastian) [inline firebase-firestore/src/test/java/com/google/firebase/firestore/remote/RemoteSerializerTest.java:406]: Super optional nit (here and below): To me, `set` is the base case and `patch` is more advanced. I would flip the order of these assertions.

**Review by schmidt-sebastian — COMMENTED** [review]
Very nice!

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalSerializer.java [hunk 2]
```diff
@@ -171,6 +174,35 @@ MutationBatch decodeMutationBatch(com.google.firebase.firestore.proto.WriteBatch
...
+        batch =
+            com.google.firebase.firestore.proto.WriteBatch.newBuilder(batch)
+                .removeWrites(i)
+                .removeWrites(i - 1)
+                .addWrites(i - 1, newMutationBuilder.build())
+                .build();
+        --i;
+      }
+    }
+
```
[the remark this answers is not in the document]
Comment (thebrianchen) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalSerializer.java:202]: Done. Used a builder to accumulate everything before building at the end of the loop

[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalSerializer.java:202]: It looks like prepending an element to a list in Protobuf is an O(n) operation. Would it be possible to reverse this loop so we can append at the end? We could also just scrap the Builder entirely and just build up `mutations` directly (see the 5 lines below).

[the remark this answers is not in the document]
Comment (thebrianchen) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalSerializer.java]: ended up using mutations directly.

**Review by schmidt-sebastian — APPROVED** [review]

**Pull request merged**